api/src/com/cloud/network/vpc/OSPFZoneConfig.java

```diff
@@ -0,0 +1,332 @@
+
+    public Short getRetransmitInterval() {
+        return retransmitInterval;
+    }
+
+    public void setRetransmitInterval(Short retransmitInterval) {
+        this.retransmitInterval = retransmitInterval;
+    }
+
+    public Short getTransitDelay() {
+        return transitDelay;
+    }
+
+    public void setTransitDelay(Short transitDelay) {
+        this.transitDelay = transitDelay;
+    }
+
+    public Authentication getAuthentication() {
+        return authentication;
+    }
+
+    public void setAuthentication(Authentication authentication) {
+        this.authentication = authentication;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public String[] getSuperCIDR() {
+        return superCIDRList;
+    }
+
+    public void setSuperCIDR(String[] superCIDR) {
+        this.superCIDRList = superCIDR;
+    }
+
+    public Boolean getEnabled() {
+        return enabled;
+    }
+
+    public void setEnabled(Boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    public enum Params {
+        PROTOCOL, AREA, HELLO_INTERVAL, DEAD_INTERVAL, RETRANSMIT_INTERVAL, TRANSIT_DELAY, AUTHENTICATION, SUPER_CIDR, PASSWORD, ENABLED
+    }
+
+    public enum Protocol {
+        OSPF, BGP
+    }
+
+    public enum Authentication {
+        MD5, PLAIN_TEXT
+    }
+
+    public OSPFZoneConfig() {
+    }
+
+    public void setDefaultValues(final Map<String, String> details) {
+        this.protocol = details.get(Params.PROTOCOL.name()) == null ? Protocol.OSPF : Protocol.valueOf(details.get(Params.PROTOCOL.name()));
+        this.ospfArea = details.get(Params.AREA.name()) == null ? "0" : details.get(Params.AREA.name());
+        this.helloInterval = Short.valueOf(details.get(Params.HELLO_INTERVAL.name()) == null ? "10" : details.get(Params.HELLO_INTERVAL.name()));
+        this.deadInterval = Short.valueOf(details.get(Params.DEAD_INTERVAL.name()) == null ? "40" : details.get(Params.DEAD_INTERVAL.name()));
+        this.retransmitInterval = Short.valueOf(details.get(Params.RETRANSMIT_INTERVAL.name()) == null ? "5" : details.get(Params.RETRANSMIT_INTERVAL.name()));
+        this.transitDelay = Short.valueOf(details.get(Params.TRANSIT_DELAY.name()) == null ? "1" : details.get(Params.TRANSIT_DELAY.name()));
+        this.authentication = details.get(Params.AUTHENTICATION.name()) == null ? Authentication.MD5
+                : Authentication.valueOf(details.get(Params.AUTHENTICATION.name()));
+        this.password = details.get(Params.PASSWORD.name()) == null ? "" : details.get(Params.PASSWORD.name());
+        String cidr_list = details.get(Params.SUPER_CIDR.name()) == null ? "200.100.0.0/20" : details.get(Params.SUPER_CIDR.name());
+        this.superCIDRList = cidr_list.split(",");
+        this.enabled = details.get(Params.ENABLED.name()) == null ? Boolean.FALSE : Boolean.valueOf(details.get(Params.ENABLED.name()));
```

Comment:
1. This ternary operation is repeated throughout this method.  Please extract to a common utility method such as ``getMapValueWithDefault`` and include unit tests.
1. Since the value being retrieved is a ``String``, test the value with ``Strings.isNullOrEmpty``to protect against both null and blank values.